lib/index.js

```diff
@@ -33,7 +33,8 @@ module.exports = function rewireify(file, options) {
 
   function end() {
     post += "/* This code was injected by Rewireify */\n";
-    post += "if (Object.isExtensible(module.exports)) {\n";
+    post += "if ('functionobject'.match(typeof module.exports) && \n"
```

Comment:
Sorry, this code is a bit unreadable because I'm trying to fit it into a string.

`match` returns null if the owning string does not satisfy the regex that's passed in. Otherwise it returns a truthy value. Strings are coerced to regexes.

`'functionobject'` matches the result of calling `typeof` on a function or object (including arrays), but not of any primitives.

If we extended your suggested code to check for `'[object Function]'`, that would also work. I could write a function that performs this and then `toString` if that's preferable.